sdk/trace/provider.go

```diff
@@ -180,7 +180,7 @@ func (p *TracerProvider) UnregisterSpanProcessor(sp SpanProcessor) {
 	if p.isShutdown {
 		return
 	}
-	old := p.spanProcessors.Load().(spanProcessorStates)
+	old := *(p.spanProcessors.Load())
```

Comment:
how about extracting `*(p.spanProcessors.Load())` to a method e.g. named `getSpanProcessors` 😅